### PR TITLE
Fermi net nico

### PR DESCRIPTION
**Pull request by NicoRenaud**
This is not a working code just the skeleton.
The idea is to refact FermiNet so that we can speed up the calculation. The only file is `fermi_nico.py`

The main idea is to define each individual layer. So we have :

- a preprocess layer that concatenate the electronic positions
- a class for a single individual layer that takes hi and hij as input and returns new hi and hij
- and an orbital layer that takes hi as input and should return a (nbatch, ndet, nelec, norb) tensor.

I haven't figured out the exponential envelopes yet for the orbital but it shouldn't be too difficult. However the `w * h + g` is now just a tensor prod and an addition

@Rik-Breebaart do you think  this could work ?

### CODE REVIEW

File: qmctorch/wavefunction/fermi_nico.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+class FermiOrbital(nn.Module):
+
+    def __init__(self, mol, Kdet, input_size):
+        """Computes all the orbitals from the output of the last layer."""
+
+        self.mol = mol
+        self.Kdet = Kdet
+
+        self.W = nn.Parameters(torch.rand(
```
Comment (Rik-Breebaart) [inline qmctorch/wavefunction/fermi_nico.py:148]: This would atleast remove the K linear layers for the different determinant. But this would still need to be used Nelec times to create the different orbitals. If we can change the exponential weight $\mathbf{\Sigma}^{k\alpha}_{im} \in \mathbb{R}^{3x3}$ to a $\mathbb{R}^{Kx3x3}$ and then normalize this to K exponents and do this for each atom. Resulting in a KxNatom exponents which can then be put in a linear layer with KxNatom weights which could give a vector of size K. which can be multiplied by the linear output. However this would require that nn.parameters can give a 3d weights vector. If that is possible then this should definitly work.
